src/offset.js

```diff
@@ -93,16 +93,24 @@ jQuery.fn.extend({
 
 		docElem = doc.documentElement;
 
-		// Make sure it's not a disconnected DOM node
-		if ( !jQuery.contains( docElem, elem ) ) {
+		// Make sure it's not a disconnected DOM node,
+		// which is enough only if user-agent does not support ShadowDOM
+		if ( !elem.createShadowRoot && !jQuery.contains( docElem, elem ) ) {
```

Comment:
The [conflation of unrelated behaviors](https://github.com/jquery/jquery/issues/1784#issuecomment-71842046) is Shadow DOM support and applicability of `getBoundingClientRect` to disconnected elements. As I mentioned in #1784, I'd like to see this entire block (and all detection or special treatment of disconnection) removed.